src/Mobs/Villager.cpp

```diff
@@ -180,18 +195,63 @@ void cVillager::HandleFarmerPrepareFarmCrops()
 
 
 
-void cVillager::HandleFarmerTryHarvestCrops()
+bool cVillager::HandleFarmerTryHarvestCrops()
 {
-	// Harvest the crops if the villager isn't moving and if the crops are closer then 2 blocks.
-	if (!m_PathfinderActivated && (GetPosition() - m_CropsPos).Length() < 2)
+	// Harvest the crops if it is closer than 1 block.
+	if ((GetPosition() - m_CropsPos).Length() < 1)
 	{
 		// Check if the blocks didn't change while the villager was walking to the coordinates.
 		BLOCKTYPE CropBlock = m_World->GetBlock(m_CropsPos);
 		if (IsBlockFarmable(CropBlock) && m_World->GetBlockMeta(m_CropsPos) == 0x7)
 		{
+			m_World->BroadcastSoundParticleEffect(EffectID::PARTICLE_BLOCK_BREAK, m_CropsPos, m_World->GetBlock(m_CropsPos));
 			m_World->DropBlockAsPickups(m_CropsPos, this, nullptr);
 			m_ActionCountDown = 20;
+			return false;  // The block no longer exist since he broke it.
 		}
+		return false;  // The block is no longer farmable.
+	}
+	return true;  // Keep walking.
+}
+
+
+
+
+
+void cVillager::CheckForNearbyCrops()
+{
+
+	// Search for adjacent crops
+
+	Vector3i North = {0, 0, -1};
+	Vector3i South = {0, 0, 1};
+	Vector3i East = {1, 0, 0};
+	Vector3i West = {-1, 0, 0};
+
+	std::vector<Vector3i> Directions = {North, South, East, West};
```

Comment:
```suggestion
	constexpr std::array<Vector3i, 4> Directions = {{0, 0, -1}, {0, 0, 1}, {1, 0, 0}, {-1, 0, 0}};
```

This should be better - please check if the constexpr actually compiles